detekt-gradle-plugin/src/intTest/kotlin/io/gitlab/arturbosch/detekt/DetektXmlReportMergeTest.kt

```diff
@@ -0,0 +1,106 @@
+            |  output = project.layout.buildDirectory.file("reports/detekt/merge.xml")
+            |}
+            |
+            |subprojects {
+            |  ${groovy.gradleSubprojectsApplyPlugins}
+            |  
+            |  detekt {
+            |    reports.xml.enabled = true
+            |  }
+            |  
+            |  plugins.withType(io.gitlab.arturbosch.detekt.DetektPlugin) {
+            |    tasks.withType(io.gitlab.arturbosch.detekt.Detekt) { detektTask ->
+            |       xmlReportMerge.configure { mergeTask ->
+            |         mergeTask.mustRunAfter(detektTask)
+            |         mergeTask.input.from(detektTask.xmlReportFile)
+            |       }
+            |    }
+            |  }
+            |}
+            |""".trimMargin()
+        val kotlin = DslTestBuilder.kotlin()
+        val kotlinBuildFileContent = """
+            |${kotlin.gradlePlugins}
+            |
+            |allprojects {
+            |  ${kotlin.gradleRepositories}
+            |}
+            |
+            |val xmlReportMerge by tasks.registering(io.gitlab.arturbosch.detekt.report.XmlReportMergeTask::class) {
+            |  output.set(project.layout.buildDirectory.file("reports/detekt/merge.xml"))
+            |}
+            |
+            |subprojects {
+            |  ${kotlin.gradleSubprojectsApplyPlugins}
+            |  
+            |  detekt {
+            |    reports.xml.enabled = true
+            |  }
+            |  
+            |  plugins.withType(io.gitlab.arturbosch.detekt.DetektPlugin::class) {
+            |    tasks.withType(io.gitlab.arturbosch.detekt.Detekt::class) detekt@{
+            |       xmlReportMerge.configure {
+            |         this.mustRunAfter(this@detekt)
+            |         input.from(this@detekt.xmlReportFile)
+            |       }
+            |    }
+            |  }
+            |}
+            |""".trimMargin()
+
+        listOf(
+            groovy to groovyBuildFileContent,
+            kotlin to kotlinBuildFileContent
+        ).forEach { (builder, mainBuildFileContent) ->
+            it("using ${builder.gradleBuildName}") {
+                val projectLayout = ProjectLayout(numberOfSourceFilesInRootPerSourceDir = 0).apply {
+                    addSubmodule(
+                        name = "child1",
+                        numberOfSourceFilesPerSourceDir = 2,
+                        numberOfCodeSmells = 2
+                    )
+                    addSubmodule(
+                        name = "child2",
+                        numberOfSourceFilesPerSourceDir = 4,
+                        numberOfCodeSmells = 4
+                    )
+                }
+
+                val gradleRunner = DslGradleRunner(projectLayout, builder.gradleBuildName, mainBuildFileContent)
+                gradleRunner.setupProject()
+                gradleRunner.runTasksAndExpectFailure("detekt", "xmlReportMerge", "--continue") { result ->
```

Comment:
```suggestion
                gradleRunner.runTasksAndExpectFailure("detekt", "xmlReportMerge", "--continue", "--stacktrace") { result ->
```

That's a nit. Also I'm unsure you need the `--continue`.

Reply:
Ah good call. I have left my debugging code here.